File: src/BukkitPE/event/plugin/PluginEvent.php
<?php
/**
 * Events related Plugin enable / disable events
 */
namespace BukkitPE\event\plugin;

use BukkitPE\event\Event;
use BukkitPE\plugin\Plugin;


abstract class PluginEvent extends Event{

	/** @var Plugin */
	private $plugin;

	public function __construct(Plugin $plugin){
		$this->plugin = $plugin;
	}

	/**
	 * @return Plugin
	 */
	public function getPlugin(){
		return $this->plugin;
	}
}

File: src/net/BukkitPE/event/plugin/PluginEvent.java
package net.BukkitPE.event.plugin;

import net.BukkitPE.event.Event;
import net.BukkitPE.event.HandlerList;
import net.BukkitPE.plugin.Plugin;

/**
 * author: MagicDroidX
 * BukkitPE Project
 */
public class PluginEvent extends Event {

    private static final HandlerList handlers = new HandlerList();

    private Plugin plugin;

    public PluginEvent(Plugin plugin) {
        this.plugin = plugin;
    }

    public static HandlerList getHandlers() {
        return handlers;
    }

    public Plugin getPlugin() {
        return plugin;
    }
}
